### PR TITLE
Add self-hosted renovate

### PR DESCRIPTION
**Pull request by noahtallen**
Some time ago, Renovate was suspended at the organization level due to a security issue. While that was fixed, the account remains suspended. This PR proposes moving our config to "self-hosted" via GitHub actions, inspired by Jetpack. (https://github.com/Automattic/jetpack/pull/23307)

I copy/pasted the config file from Jetpack (https://github.com/Automattic/jetpack/blob/trunk/.github/workflows/renovate.yml), and added the same RENOVATE_TOKEN secret to wp-calypso. (See pMz3w-eQ7-p2#comment-96792) I did make some small changes as well.

## Testing:

Check github workflow run!

image

### CODE REVIEW

**Comment by noahtallen** [comment]
See the new dashboard: https://github.com/Automattic/wp-calypso/issues/80201

File: renovate.json5
Comment (noahtallen) [inline renovate.json5:1]: Moved this to json5 because comments are super helpful to have

File: renovate.json5 [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			groupName: 'redux-related packages',
+			matchPackagePatterns: [ 'redux' ],
+			matchPackageNames: [ 'react-redux' ],
+			prPriority: 2,
+		},
+		{
+			groupName: 'typescript-related packages',
+			matchPackagePatterns: [ 'typescript-eslint' ],
+			matchPackageNames: [ 'typescript' ],
+			prPriority: 2,
```
Comment (dpasque) [inline renovate.json5:31]: **Question (non-blocking):** Mostly out of curiosity, why don't we want `@types` packages included in the renovate update PRs? Have we had issues with them getting too far ahead?

**Comment by dpasque** [comment]
Overall, this change looks good to me! 😄 I haven't done too much with renovate bot in the past, so I might be missing some context about some of the package update rules, and it might be good to have someone from frameworks double-check. 

Thanks for doing this, that's awesome that we'll get renovate running again! 🥳

File: renovate.json5 (hunk 1 above)
Comment (noahtallen) [reply renovate.json5:31]: We do want `@types` to update! It just doesn't make sense to update them with Typescript -- they should be updated when the corresponding package is updated. For example, `@types/react` probably only needs to be updated when `react` is updated.

By the way, the config automatically updates everything by default when we don't specify a certain package (unless we manually ignore it). The groups we set up just add our own own settings on top of that, to help us be more organized.

File: renovate.json5 [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	// Create PRs immediately. Extra branches we don't see still consume CI resources
+	// when they are created/rebased, so we want visibility into them quickly.
+	prCreation: 'immediate',
+	internalChecksFilter: 'strict',
+	// Don't rebase PRs constantly, to avoid using too much CI.
+	rebaseWhen: 'conflicted',
+
+	// --- Metadata settings for git ---
+	labels: [ 'Framework', '[Type] Task' ],
+	reviewers: [ 'team:team-calypso-platform' ],
```
Comment (tyxla) [inline renovate.json5]: Totally feel free to add our team as reviewers, too. Many of those packages are actually considered framework packages and we're interested to keep them up to date.

```suggestion
	reviewers: [ 'team:team-calypso-platform', 'team:team-calypso-frameworks' ],
```

File: renovate.json5 [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	schedule: [ 'every weekend' ],
+	// Create PRs immediately. Extra branches we don't see still consume CI resources
+	// when they are created/rebased, so we want visibility into them quickly.
+	prCreation: 'immediate',
+	internalChecksFilter: 'strict',
+	// Don't rebase PRs constantly, to avoid using too much CI.
+	rebaseWhen: 'conflicted',
+
+	// --- Metadata settings for git ---
+	labels: [ 'Framework', '[Type] Task' ],
```
Comment (tyxla) [inline renovate.json5]: I think the `dependencies` label also makes sense here:

```suggestion
	labels: [ 'Framework', '[Type] Task', 'dependencies' ],
```

**Review by tyxla — APPROVED** [review]
Awesome to see this! Left a few minor suggestions, but I think it's good to go!

Let's give it a shot 🚀

**Comment by noahtallen** [comment]
Great suggestions; added!

**Pull request merged**